### ISSUE TITLE
Tagify in mode="select" seem to return different value-array after dropdown filtering

### ISSUE THREAD

**Issue by Tagirijus** [yairEO/tagify#866]
Hey again,

I just came across something and I am not sure if it might be a bug or by design: let's say I have set up a tagify object in select mode. Now assigning a custom function which will be called when the user selects something from the dropdown, would try to access something different than the `value` key of such an array: maybe the `comment` or `id` field. Hopefully my sample code will be a good working one ... did not test this specific one, though; sorry:

```js
var element = document.getElementById('initMe');
var tagify_element = new Tagify(
	element,
	{
		mode: 'select',
		whitelist: [
			{'value':'Item B','comment':'is great','id':3},
			{'value':'Item A','comment':'is also great','id':2},
			AND MORE ITEMS HERE (at least 50 or so) ...
		],
		dropdown: {
			maxItems: 10
		}
	}
);
tagify_element.on('dropdown:select', function (e) {
	console.log(tagify_element.value.id);
});
```

Now the JS console log won't be show the id, **if** I typed in the search field to narrow down the search. It seems like the value is correct, if it is one of the first 10 items _(maybe it has something to do with maxItems: 10 ?)_, but not if it's an item past these first 10.

Am I doing or understanding something wrong here? Thanks for your help. Hopefully you are able to understand and reproduce or so. Otherwise tell me, if I can come up with more information or tests. (=

**Comment by yairEO** [comment]
Hi, I cannot seem to reproduce:

https://jsbin.com/yebicin/2/edit?html,js,output

**Comment by Tagirijus** [comment]
Oh cool, thanks for the link. I modified your example and hopefully you will see now:

https://jsbin.com/tilapinuma/edit?html,js,output

Type in `d` and chose the item. The `id` key won't be accessible.

Can you "reproduce" now? Or can you at least tell me, if I am doing something wrong here?

Thanks so much for you help (and this great tool!). (=

**Comment by yairEO** [comment]
Please explain in detail what does it mean *"The id key won't be accessible."*

**Comment by Tagirijus** [comment]
Wow, I just re-tested the jsbin link and suddenly it works now there in the scribble like expected.

Clarification what happens locally on my end in my project: I have the whitelist with maybe 500 items, while the dropdown maxItems is set to 10 so that it works smoother and does not show a too big list. A single list item in the whitelist has more than just the _value_ item.

So let's say we have one item maybe something like `{'value':'Name of item','id':1,'additional_info':'Even more info about this item','searchBy':'first'}`. Now when this item is not in the first 10 items (while maxItem==10) and I type _"first"_ so that it gets filtered and shows up and then I select it: **the selected item only has {'value':'Name of item'} as a result-array in JS**. So what I meant with _"The id key won't be accessible"_ is that the selected value item has only the _"value"_ key of the "result" array accessible, but not the _'id'_ nor the _'additional_ino'_ etc.

Maybe I will be able to alter the jsbin example so that it is reproducible. Thanks for your help already!

**Comment by Tagirijus** [comment]
I found something out. And it gets very weird. Maybe it has something to do with choosing the element via keyboard, when it is not in the first 3 elements. Watch my gif:

![Tagify maxItems Issue](https://i.imgur.com/9tFwBHj.gif)

First I click every of the three items. Id is in the array. Then I search fot the _Item d_ item and **click** it to select it. Id is in the array as well. But then I search for the item again, but press down and enter this time: no `id` in the array. Can you confirm?

**Comment by Tagirijus** [comment]
Oh ... now I figured out something else, which might be quite important: It's not mainly about getting this array-value in JS, **but in the POST body**, when submitting the form and trying to access the value from PHP!

So as far as I can tell it seems to have something to do with:

- mode == select
- maxItem is set and the chosen item has not to be in the first items, which are shown (so one have to search it and then select it)
- it's about the value which gets send via POST, when submitting a form _(or as seen in the above GIF it also happens in JS)_
- the issue only occurs when selecting the item via keyboard (not clicking)

So let's say I choose _Item d_ in the select element with my keyboard and press _"submit"_ in the form. The JSON string which would be delivered for this input element would only contain _"{'value':'Item d'}"_ then and nothing else.

**Comment by yairEO** [comment]
I admire your curiosity an enthusiasm to find the root cause of the bug!

I can very it also happens only on keyboard selection, and will debug Tagify's source code to find what is causing this.

**Comment by Tagirijus** [comment]
For such good tools / plugins / frameworks /etc. I am always eagger to make thigns better and help out. (=

Thanks so much for looking into this!

**Comment by Tagirijus** [comment]
Thanks for this fix as well. Totally looking forward to the next release then, yay. (=

**Comment by yairEO** [comment]
Thanks :) released yesterday `4.6.0`

**Comment by Tagirijus** [comment]
Cool, very fast fix! Thanks for that. Seems to work now. (=
